packages/kit/src/runtime/server/page/load_node.js

```diff
@@ -98,8 +98,12 @@ export async function load_node({
 				let response;
 
 				if (/^[a-zA-Z]+:/.test(url)) {
-					// external fetch
```

Comment:
this comment shouldn't be removed

also, I would expect the user-provided function to be used in all cases and not just the external case

Reply:
> this comment shouldn't be removed

sure

> also, I would expect the user-provided function to be used in all cases and not just the external case

for internal case you can have endpoint call instead true request. Basically, it's has sense only for external API.

Reply:
> also, I would expect the user-provided function to be used in all cases and not just the external case

This confused me for a while too. But I think the idea is that, in `load`, you would continue to call `fetch` with a path-only URI, which would then call your `handle` hook (without it being an additional network request), and then _there_ in your `handle` you would manually call `fetch` with whatever the appropriate full URL would be - and that you wouldn't really need the `serverFetch` hook at all in that case.

Reply:
Put another way, the existing `handle` hook is essentially already the `fetch`/`serverFetch` hook for same-origin requests. We don't need another one for that.

Reply:
It seems a bit inconsistent though that you'd use `serverFetch` to alter the request URL for external requests and `handle` to alter the request URL for internal requests. I think it'd be a clearer API for `serverFetch` to be available regardless

Also, imagine that `/api/endpoint` is on the same domain, but on another server running in another language. In that case you may want to use `serverFetch` to do something like hit the service directly instead of the external load balancer. In that case you don't necessarily want to write more complicated code to have `handle` proxy to the other server compared when it'd be much simpler to have `serverFetch` simply change the URL

Reply:
How would `handle` and `serverFetch` interact in that case? Without calling `handle` first, how would Kit know whether this request is going to be handled by an internal page/endpoint or an external one?  And if it does call `handle` first, how does the handoff to `serverFetch` happen?

Reply:
`handle` wouldn't be called at all because you wouldn't be hitting a SvelteKit endpoint. If `serverFetch` changes the URL from `/api/endpoint` to `http://backend-service/api/endpoint` it should be pretty clear that it's external since it's a different host

Reply:
or even request by unix socket

Reply:
> I think it'd be a clearer API for serverFetch to be available regardless

hmm, I suppose serverFetch has sense only for external API, because for internal API SvelteKit carries very well. 
Also, in that case, users should provide this check inside serverFetch somehow in any case.